app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/StartController.scala

```diff
@@ -88,17 +88,15 @@ class StartController @Inject()(
     }
   }
 
-  def returnAfterMapping(id: Option[String] = None): Action[AnyContent] = Action.async { implicit request =>
+  def returnAfterMapping(): Action[AnyContent] = Action.async { implicit request =>
     withSubscribingAgent { agent =>
+      val sjr = agent.getMandatorySubscriptionRecord
       continueUrlActions.withMaybeContinueUrlCached {
-        id match {
-          case Some(continueId) =>
-            for {
-              record <- subscriptionJourneyService.getMandatoryJourneyRecord(ContinueId(continueId))
-              _      <- subscriptionJourneyService.saveJourneyRecord(record.copy(mappingComplete = true))
-            } yield Redirect(routes.TaskListController.showTaskList())
-
-          case None => Future.successful(Redirect(routes.TaskListController.showTaskList()))
+        sjr.authProviderId flatMap { authProviderId =>
```

Comment:
Trying to figure out why you need to flatMap here (I guess it's a Future, but it's weird because sjr is not a future)?  But it compiles, so never mind :)